packages/vite/src/generators/vitest/vitest-generator.ts

```diff
@@ -62,20 +65,47 @@ export async function vitestGeneratorInternal(
   tasks.push(initTask);
   tasks.push(ensureDependencies(tree, schema));
 
-  const nxJson = readNxJson(tree);
-  const hasPluginCheck = nxJson.plugins?.some(
-    (p) =>
-      (typeof p === 'string'
-        ? p === '@nx/vite/plugin'
-        : p.plugin === '@nx/vite/plugin') || hasPlugin
-  );
-  if (!hasPluginCheck) {
-    const testTarget = schema.testTarget ?? 'test';
-    addOrChangeTestTarget(tree, schema, testTarget);
-  }
+  addOrChangeTestTarget(tree, schema, hasPlugin);
 
   if (!schema.skipViteConfig) {
-    if (schema.uiFramework === 'react') {
+    if (schema.uiFramework === 'angular') {
```

Comment:
It is conceptually challenging.

We could implement heuristics that analyze the imports. The problem is that a project can be defined with a "targeted" type before it has anything in it that is specific to that framework.

Let's say I have an Angular library with nothing Angular-specific yet, it would assume `uiFramework: none`, and then simply not work once I decide to add something Angular-specific.
Then we would need a generator that updates the configuration according to the detected framework so things can get a bit complicated.

Reply:
Agreed. 

With angular applications and buildable libraries, we can check the project graph to see if the project has a dependency on any `@angular` or `@nx/angular` package. 

If they generated their library via `@nx/angular:library` generator, one of those dependencies should still exist. 

But if they generated their library via `@nx/js:library` then it wouldn’t.

That said, I think that is good enough.

We can _try_ to infer it if someone has not passed in `--uiFramework`.

Worse case they end up with “none” which is the same outcome as they didn’t pass in `--uiFramework`.

At the start of the generator we can add something like 

`schema.uiFramework ??= tryDetermineFramework()`

so that it only attempts it if the user has not provided it.

the `tryDetermineFramework` can use the project graph to look for dependencies.

```ts
const REACT_DEPS = [“react”, “react-dom”, …]
const ANGULAR_DEPS = [“@angular/core”, “@nx/angular”, …]
const projectGraph = await createProjectGraphAsync();

const projectDeps = projectGraph.deps.filter(dep => dep.source === schema.project).reduce((collectedDeps, dep) => collectedDeps.add(dep), new Set())

const isReact = REACT_DEPS.some(dep => projectDeps.has(dep));
const isAngular = ANGULAR_DEPS.some(dep => projectDeps.has(dep))

schema.uiFramework = isReact 
  ? “react” 
  : isAngular
    ? “angular”
     : “none”
```

Reply:
Actually, even for the rare cases where the detection would be mistaken, one could easily rerun the configuration and pass the `uiFramework` manually to overwrite the config.

I'll add the detection as soon as i can.

Reply:
Just added the framework detection based on dependencies analysis like @Coly010 suggested.